zenoh/src/net/routing/hat/linkstate_peer/network.rs

```diff
@@ -247,7 +248,7 @@ impl Network {
             whatami: self.graph[idx].whatami,
             locators: if details.locators {
                 if idx == self.idx {
-                    Some(self.runtime.get_locators())
+                    Some(self.runtime.upgrade().unwrap().get_locators())
```

Comment:
It is mostly done to be able to detect issues with arc-weak management. It would be way more complicated to detect them if I return just an empty result. In this not being able to acquire a strong runtime is likely a non-recoverable error.

Reply:
Maybe then use `debug_assert!` and `log::error!` with "internal error" message? This will allow to detect the problem and avoid panic. Anyway this may happen on shutdown stage only. I think it's better to shutdown with error message than abort the whole user application

Reply:
It will allow to detect the problem, but will not give you stack trace and a such it will severely complicate the debugging.
If we agree that this is an impossible situation (which it should be, given that runtime is expected to terminate all the tasks upon its close) I suggest we just panic and not overload the code with unnecessary if/match statements and error logs, which likely nobody will ever look at.